Xamarin.Forms.Platform.iOS/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,249 @@
+			{
+				var tableView = view as UITableView;
+
+				if (!_set)
+				{
+					_origininalY = tableView.ContentOffset.Y;
+					_set = true;
+				}
+
+				if (_origininalY < 0)
+					return true;
+
+				if (refreshing)
+					tableView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY - _refreshControl.Frame.Size.Height), true);
+				else
+					tableView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY), true);
+				return true;
+			}
+
+			if (view is UICollectionView)
+			{
+				var collectionView = view as UICollectionView;
+
+				if (!_set)
+				{
+					_origininalY = collectionView.ContentOffset.Y;
+					_set = true;
+				}
+
+				if (_origininalY < 0)
+					return true;
+
+				if (refreshing)
+					collectionView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY - _refreshControl.Frame.Size.Height), true);
+				else
+					collectionView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY), true);
+				return true;
+			}
+
+			if (view is UIWebView)
+			{
+				return true;
+			}
+
+			if (view is UIScrollView)
+			{
+				var uiScrollView = view as UIScrollView;
+
+				if (!_set)
+				{
+					_origininalY = uiScrollView.ContentOffset.Y;
+					_set = true;
+				}
+
+				if (_origininalY < 0)
+					return true;
+
+				if (refreshing)
+					uiScrollView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY - _refreshControl.Frame.Size.Height), true);
+				else
+					uiScrollView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY), true);
+				return true;
+			}
+
+			if (view.Subviews == null)
+				return false;
+
+			for (int i = 0; i < view.Subviews.Length; i++)
+			{
+				var control = view.Subviews[i];
+				if (TryOffsetRefresh(control, refreshing))
+					return true;
+			}
+
+			return false;
+		}
+
+		bool TryInsertRefresh(UIView view, int index = 0)
+		{
+			_refreshControlParent = view;
+
+			if (view is UITableView || view is UICollectionView)
+			{
+				view.InsertSubview(_refreshControl, index);
+				return true;
+			}
+
+			if (view is UIWebView)
+			{
+				var uiWebView = view as UIWebView;
+				uiWebView.ScrollView.InsertSubview(_refreshControl, index);
+				return true;
+			}
+
+			if (view is UIScrollView)
+			{
+				var uiScrollView = view as UIScrollView;
+				view.InsertSubview(_refreshControl, index);
+				uiScrollView.AlwaysBounceVertical = true;
+				return true;
+			}
+
+			if (view.Subviews == null)
+				return false;
+
+			for (int i = 0; i < view.Subviews.Length; i++)
+			{
+				var control = view.Subviews[i];
+				if (TryInsertRefresh(control, i))
+					return true;
+			}
+
+			return false;
+		}
+
+		void UpdateColors()
+		{
+			if (Element == null || _refreshControl == null)
+				return;
+
+			if (Element.RefreshColor != Color.Default)
+				_refreshControl.TintColor = Element.RefreshColor.ToUIColor();
+
+			SetBackgroundColor(Element.BackgroundColor);
+		}
+
+		void UpdateIsRefreshing()
+		{
+			IsRefreshing = Element.IsRefreshing;
+		}
+
+		void UpdateIsEnabled()
```

Comment:
So many curly braces in this method. Maybe we could remove some to make it smaller.